wandb/errors/__init__.py

```diff
@@ -19,31 +22,35 @@ def __init__(self, message, context: Optional[dict] = None) -> None:
         if context:
             self.context = context
 
-    # For python 2 support
-    def encode(self, encoding):
-        return self.message
-
 
 class CommError(Error):
-    """Error communicating with W&B"""
+    """Error communicating with W&B (legacy error for backwards compatibility)"""
 
     def __init__(self, msg, exc=None) -> None:
-        super().__init__(msg)
-        self.message = msg
         self.exc = exc
+        self.message = msg
+        super().__init__(self.message)
 
 
-class UsageError(Error):
-    """API Usage Error"""
+class BackendError(CommError):
+    """Error communicating with W&B backend"""
 
-    pass
+
+class UsageError(Error):
+    """Raised when an invalid usage of the SDK API is detected"""
 
 
 class UnsupportedError(UsageError):
     """Raised when trying to use a feature that is not supported"""
 
 
+class DependencyError(UsageError):
+    """Raised when there is a missing or invalid dependency"""
```

Comment:
Lets remove (or comment out) the errors that arent used yet, in protobuf and here.   might be nice to keep them as comments so we have the idea and develop it over time.